src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx

```diff
@@ -179,7 +183,54 @@ const AuthenticatorButton: UISchemaElementComponent<{
         >
           {label}
         </Typography>
-        {description && (
+        {gracePeriodRequiredDescription && gracePeriodExpiry && (
+          <Box
+            sx={{
+              display: 'flex',
+              margin: tokens.Spacing0,
+              marginBlockStart: tokens.Spacing1,
+            }}
+          >
+            <ClockIcon
+              sx={{
+                color: tokens.TypographyColorWarning,
+              }}
+            />
+            <Box
+              sx={{
+                marginLeft: tokens.Spacing1,
+              }}
+            >
+              <Typography
+                paragraph
+                data-se="authenticator-grace-period-required-description"
+                sx={{
+                  fontSize: tokens.TypographySizeSubordinate,
+                  fontWeight: tokens.TypographyWeightBodyBold,
+                  color: tokens.TypographyColorWarning,
+                  margin: tokens.Spacing0,
+                  textAlign: 'start',
+                }}
+              >
+                {gracePeriodRequiredDescription}
+              </Typography>
+              <Typography
+                paragraph
+                data-se="authenticator-grace-period-expiry-date"
+                sx={{
+                  fontSize: tokens.TypographySizeSubordinate,
+                  fontWeight: tokens.TypographyWeightBody,
+                  color: tokens.TypographyColorSubordinate,
+                  margin: tokens.Spacing0,
+                  textAlign: 'start',
+                }}
+              >
+                {gracePeriodExpiry}
+              </Typography>
+            </Box>
+          </Box>
+        )}
+        {!gracePeriodRequiredDescription && description && (
```

Comment:
Can this logic be refactored out so they are implicitly mutually exclusive rather than explicitly? (e.g. using `if/else` or moving to a function named something like `renderDescription()` that returns a component)

Reply:
Yes, moved it out to a `renderDescription()` with `if/else if`